deploy_heroku.py

```diff
@@ -3,61 +3,119 @@
 import json
 import os
 import subprocess
+import urllib.parse
 import urllib.request
-from typing import List
+from typing import List, Optional
 
 
 @dataclasses.dataclass
 class HerokuRelease:
     id: str
     version: int
     status: str
+    description: str
+    slug_id: str
+
+
+class DeploymentType:
+    ROLLBACK = "rollback"
+    REDEPLOY = "redeploy"
+
+
+def convert_response(data) -> HerokuRelease:
+    return HerokuRelease(
+        **{prop.name: data[prop.name] for prop in dataclasses.fields(HerokuRelease) if prop.name != "slug_id"},
+        slug_id=data["slug"]["id"],
+    )
+
+
+def get_request(app: str, api_key: str, endpoint: str, data: Optional[bytes] = None) -> urllib.request.Request:
```

Comment:
```suggestion
def get_request(app: str, api_key: str, endpoint: str, data: Optional[dict] = None) -> urllib.request.Request:
```

Reply:
Bei dem `Request` muss data `bytes` sein.

https://docs.python.org/3/library/urllib.request.html#urllib.request.Request

Deshalb hatte ich das so direkt in die Funktion reingegeben. Ich finde `urllib` umständlich... :(

Reply:
Sorry, ich habe das Kommentar unabsichtig einzeln abgeschickt und jetzt habe ich aber der Rest fertig - hoffe du verstehst jetzt was ich meine...

Reply:
Okay, verstsanden.